### PR TITLE
Fix all the things

### PR DESCRIPTION
**Pull request by lolPants**
# Docker Image
Until this gets merged into the main repo, my changes will be available on my own docker account.  
`docker pull lolpants/dockercraft`

# Changes
I've made a few changes and bugfixes, have a read.

## Download URL
The Docker download URL changed causing the Go server to fail on download.
This updates for the new URL and also adds root CA certs for the Debian intermediary container to allow the new download to succeed.

## Web Admin
This was added but I reverted it due to potential security risks. Might revisit it at some point.

## Authentication
By default the Cuberite server doesn't care if you're logged in. This can be a problem if you want to use a whitelist. I've turned that on. *(Also makes player skins work)*

## Dependencies
Both Cuberite and Go were out of date. They're now on the latest versions at time of writing. They've both been tested and work. This means you can now login to Dockercraft from Minecraft versions `1.8` to `1.12`.

## New Containers
Starting a new container will cause all other containers within Dockercraft to disappear. They're still running, but they don't exist in the world until they're updated. Using a hack this forces all containers to re-render when this happens and fixes that issue.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -2,22 +2,23 @@ FROM alpine:3.6 AS wget
 RUN apk add --no-cache ca-certificates wget tar
 
 FROM wget AS docker
-ARG DOCKER_VERSION=17.04.0-ce
-RUN wget -qO- https://get.docker.com/builds/Linux/x86_64/docker-${DOCKER_VERSION}.tgz | \
+ARG DOCKER_VERSION=17.09.0-ce
+RUN wget -qO- https://download.docker.com/linux/static/stable/x86_64/docker-${DOCKER_VERSION}.tgz | \
```
Comment (thaJeztah) [inline Dockerfile]: You could even skip this entire build-stage; the `COPY --from` syntax also allows copying from an image. The following would pulls the official `docker:17.09` image, and copy the CLI binary from that;

```Dockerfile
COPY --from=docker:17.09 /usr/local/bin/docker /bin
```

File: setup.go [hunk 2]
```diff
@@ -14,7 +14,7 @@ import (
 )
 
 const (
-	downloadURL   = "https://get.docker.com/builds/Linux/x86_64/docker-"
+	downloadURL   = "https://download.docker.com/linux/static/stable/x86_64/docker-"
```
Comment (thaJeztah) [inline setup.go]: Also wondering if the whole version-compare in `GetDockerBinary()` should still be there; versions older than Docker-CE 17.09 have reached EOL, so perhaps it should be simplified to remove that code.

**Comment by scoobyshi** [comment]
Would also love to see this PR merged, and confirm it resolves issues running Dockercraft on MacOS as well.

**Comment by dave-tucker** [comment]
LGTM. Thanks for the contribution @lolPants

**Pull request merged**